lib/rubygems/core_ext/kernel_require.rb

```diff
@@ -18,7 +18,7 @@ module Kernel
   end
 
   file = Gem::KERNEL_WARN_IGNORES_INTERNAL_ENTRIES ? "<internal:#{__FILE__}>" : __FILE__
-  module_eval <<'RUBY', file, __LINE__
+  module_eval <<'RUBY', file, __LINE__ + 1
```

Comment:
Oh, nice catch! Since we almost missed this regression, should we add a test where an exception is raised during a require and check that the backtrace is correct?

Reply:
It seems difficult, because we would have to hardcode some line numbers (e.g. the line of `raise` or the line of `def`), no?

I also noticed many other places use `__LINE__ + 1` in the codebase, so it's just the correct pattern that I unfortunately missed initially.

Reply:
Yeah, well I was thinking of a "clever" solution of reading and greping the file and inferring line numbers from that, but it's probably not worth it?

Reply:
Maybe a rubocop cop is best!

Reply:
https://github.com/rubygems/rubygems/pull/4097